controllers/util/generate.go

```diff
@@ -18,6 +19,15 @@ func GenerateNameForCephNFS(initData *ocsv1.StorageCluster) string {
 	return fmt.Sprintf("%s-cephnfs", initData.Name)
 }
 
+func GenerateNameForNFSServer(initData *ocsv1.StorageCluster) string {
+	extEndpoint := ""
+	if nfsSpec := initData.Spec.NFS; nfsSpec != nil {
+		extEndpoint = nfsSpec.ExternalEndpoint
+	}
+
+	return cmp.Or(extEndpoint, GenerateNameForNFSService(initData))
```

Comment:
```suggestion
	nfsSpec := initData.Spec.NFS
	if nfsSpec != nil && nfsSpec.ExternalEndpoint != "" {
		return nfsSpec.ExternalEndpoint
	}
	return GenerateNameForNFSService(initData)
```
anyways this is only a single comparison being done in a func and don't see the need to invoke a variadic func. wdyt?